extensions/oauth/src/main/java/org/mule/extension/oauth2/internal/DeferredExpressionResolver.java

```diff
@@ -45,4 +46,15 @@ public DeferredExpressionResolver(MuleContext muleContext) {
     }
   }
 
+  public <T> String getExpression(ParameterResolver<T> expr) {
```

Comment:
if ParameterResolver.getExpression() is not empty, then for sure is an expression and you don't need to check it again. Also, ExperssionManager should not be obtained through MuleContext